File: Example/FlexLib/TestModalVC.m
/**
 * Copyright (c) 2017-present, zhenglibao, Inc.
 * email: 798393829@qq.com
 * All rights reserved.
 *
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree.
 */



#import "TestModalVC.h"

@interface TestModalVC ()
{
    FlexModalView* modal;
}

@end

@implementation TestModalVC

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationItem.title = @"Modal Demo";
}

- (void)dealloc
{
    NSLog(@"TestModalVC dealloc");
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)tapCloseAction
{
    [self.navigationController popViewControllerAnimated:YES];
}
-(void)tapModalTop
{
    [modal setViewAttr:@"position" Value:@"top"];
    [modal showModalInView:self.view Anim:YES];
}
-(void)tapModalCenter
{
    [modal setViewAttr:@"position" Value:@"center"];
    [modal showModalInView:self.view Anim:YES];
}
-(void)tapModalBottom
{
    [modal setViewAttr:@"position" Value:@"bottom"];
    [modal showModalInView:self.view Anim:YES];
}
-(void)closeModal
{
    [modal hideModal:YES];
}
@end

File: Example/FlexSwiftDemo/TestModalVC.swift
/**
 * Copyright (c) 2017-present, zhenglibao, Inc.
 * email: 798393829@qq.com
 * All rights reserved.
 *
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

import UIKit
import FlexLib

@objc(TestModalVC)
class TestModalVC: FlexBaseVC {
    
    @objc var _modal : FlexModalView!

    override func viewDidLoad() {
        super.viewDidLoad()

        self.navigationItem.title = "Modal Demo";
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    @objc func tapModalTop()->Void{
        _modal.setViewAttr("position", value: "top")
        _modal.showModal(in: self.view, anim: true)
    }
    @objc func tapModalCenter()->Void{
        _modal.setViewAttr("position", value: "center")
        _modal.showModal(in: self.view, anim: true)
    }
    @objc func tapModalBottom()->Void{
        _modal.setViewAttr("position", value: "bottom")
        _modal.showModal(in: self.view, anim: true)
    }
}
